pkg/operator/controller/status/controller.go

```diff
@@ -264,11 +265,27 @@ func computeOperatorDegradedCondition(ingresses []operatorv1.IngressController)
 			Reason: "NoIngressControllersDegraded",
 		}
 	}
+	message := "Some ingresscontrollers are degraded:"
+	// Sort keys so that the result is deterministic.
+	keys := make([]*operatorv1.IngressController, 0, len(degradedIngresses))
+	for ingress := range degradedIngresses {
+		keys = append(keys, ingress)
+	}
+	sort.Slice(keys, func(i, j int) bool {
```

Comment:
This slice sort function is the same sort used in this PR's changes to `pkg/operator/controller/ingress/status.go`, right? 
Would it be worth it to add this sort function to `pkg/util/slice`, or somehow refactor this PR to encourage code re-use?

Reply:
The two anonymous functions are deceptively similar in appearance, but in one case, `keys` is a `[]*operatorv1.IngressController`, and in the other case, `keys` is a `[]*corev1.Pod`.  I could add a helper in something like `pkg/util/sort.go`:
```go
// ObjectLess compares two ObjectMeta values and returns true iff the first one
// is less than the second one as determined by their creation timestamps, using
// their UIDs as a tie breaker.
func ObjectLess(x, y *metav1.ObjectMeta) bool {
	if x.CreationTimestamp.Equal(&y.CreationTimestamp) {
		return x.UID < y.UID
	}
	return x.CreationTimestamp.Before(&y.CreationTimestamp)
}
```
and then use it with `sort.Slice`:
```go
	sort.Slice(keys, func(i, j int) bool {
		return oputil.ObjectLess(&keys[i].ObjectMeta, &keys[j].ObjectMeta)
	})
```
Would you prefer that to what I have now?  Is there a simpler approach that I'm not seeing?

Reply:
> Would you prefer that to what I have now? Is there a simpler approach that I'm not seeing?

The helper `ObjectLess` function is exactly what i had in mind.

Reply:
All right, latest push adds this change.